source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -330,6 +333,255 @@ void writeEllipticCurvePointFormats(const SSL_CLIENT_HELLO* ssl_client_hello,
   }
 }
 
+std::string getJA4TlsVersion(const SSL_CLIENT_HELLO* ssl_client_hello) {
+  // Check for supported_versions extension first
+  const uint8_t* data;
+  size_t size;
+  if (SSL_early_callback_ctx_extension_get(ssl_client_hello, TLSEXT_TYPE_supported_versions, &data,
+                                           &size)) {
+    CBS cbs;
+    CBS_init(&cbs, data, size);
+    uint8_t versions_len;
+    if (!CBS_get_u8(&cbs, &versions_len)) {
+      return "00";
+    }
+
+    uint16_t highest_version = 0;
+    while (CBS_len(&cbs) >= 2) {
```

Comment:
There's both a `>= 2` condition and below another `break` if we couldn't get a u16. Can you simplify it so there is only 1 loop termination condition here?